_posts/2025-03-31-zscore-hybrid-search.md

```diff
@@ -0,0 +1,178 @@
+
+### Conclusions
+
+
+Our benchmark experiments highlight the following advantages and trade-offs of Z-score normalization compared to min-max normalization in hybrid search approaches:
+
+**Search quality (measured using NDCG@10 across four datasets)**:
+
+* Z-score normalization shows a modest improvement in search quality, with an average increase of 2.08% in NDCG@10 scores.
+* This suggests that Z-score normalization may provide slightly better relevance in search results compared to the default normalization technique min-max.
+
+
+**Latency impact**:
+
+* Z-score normalization shows a small increase in latency across different percentiles, as shown in the following table:
+
+|Latency percentile	|Percent difference	|
+|---	|---	|
+|p50	|0.70%	|
+|p90	|0.50%	|
+|p99	|0.50%	|
+
+* The positive percentages indicate that Z-score normalization has slightly higher latency compared to min-max normalization, but the differences are minimal (less than 1% on average).
+
+**Trade-offs**:
+
+* There's a slight trade-off between search quality and latency. Z-score normalization offers a improvement in search relevance (2.08% increase in NDCG@10) at the cost of a marginal increase in latency (0.50% to 0.72% across different percentiles).
+
+**Consistency**:
+
+* The impact of Z-score normalization varies across datasets, with some showing more significant improvements in search quality than others.
+* Latency impact is relatively consistent across datasets, with most showing small increases or no change
+
+
+**Overall assessment**:
+
+* Z-score normalization provides a modest improvement in search quality with a negligible impact on latency.
+* The choice between Z-score and min-max normalization may depend on specific use cases, with Z-score potentially being preferred when even improvements in search relevance are valuable and the slight latency increase is acceptable.
```

Comment:
looks like some word is missing after "even"

Reply:
Good catch. Pushed the change